src/main/java/com/redhat/darcy/salesforce/DateInputField.java

```diff
@@ -0,0 +1,149 @@
+import com.redhat.darcy.ui.api.elements.Element;
+import com.redhat.darcy.ui.api.elements.Link;
+import com.redhat.darcy.ui.api.elements.Requireable;
+import com.redhat.darcy.ui.api.elements.TextInput;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * An ViewElement for a value that corresponds to a Date field on 
+ * a Salesforce object.
+ */
+public class DateInputField extends AbstractViewElement implements TextInput, 
+    DateInput, Link, Requireable {
+
+    private DateTimeFormatter formatter;
+    
+    @Require
+    private Element parent = super.parent;
+
+    @Require
+    private TextInput nestedTextInput = textInput(byInner( 
+            xpath("//span[contains(@class,'dateInput')]"),
+            htmlTag("input")));
+
+    @Require
+    private Link nestedLink = link(byInner(
+            xpath("//span[contains(@class,'dateInput')]"),
+            htmlTag("span"), htmlTag("a")));
+
+    private RequiredInput requiredInput = requiredInput(parent);
+    
+    /**
+     * A ViewElement that corresponds to a Date field on
+     * a Salesforce object.  Takes the locator returned from BySalesforce and 
+     * finds the elements for the date text input and the [today] link.
+     * 
+     * @param locator  Locator returned from BySalesforce
+     * @return DateInputField
+     */
+    public static DateInputField dateInputField(Locator locator, 
+            DateTimeFormatter formatter) {
+        return new DateInputField(locator, formatter);
+    }
+
+    public DateInputField(Locator parent, DateTimeFormatter formatter) {
+        super(parent);
+        this.formatter = formatter;
+    }
+    
+    public DateInputField(Element parent, DateTimeFormatter formatter) {
+        super(parent);
+        this.formatter = formatter;
+    }
+
+    @Override
+    public void clearAndType(CharSequence... keysToType) {
+        nestedTextInput.clearAndType(keysToType);
+    }
+
+    @Override
+    public void type(CharSequence... keysToType) {
+        nestedTextInput.type(keysToType);
+    }
+
+    @Override
+    public void clear() {
+        nestedTextInput.clear();
+    }
+
+    /** Click the date input field.  @see todayLink() for the [today] link. */
+    @Override
+    public void click() {
+        nestedTextInput.click();
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return nestedTextInput.isEnabled();
+    }
+
+    @Override
+    public String getValue() {
+        return nestedTextInput.getValue();
+    }
+
+    @Override
+    public void setDate(LocalDate date) {
+        nestedTextInput.clearAndType(date.format(formatter));
+    }
+
+    @Override
+    public LocalDate getDate() {
+        return LocalDate.parse(nestedTextInput.getValue(), formatter);
+    }
+
+    @Override
+    public boolean isRequired() {
+        return requiredInput.isDisplayed();
+    }  
+
+    /** Click the [today] link next to the date input field. */
```

Comment:
It's not supposed to be javadoc syntax.  The brackets are how it appears on the page in Salesforce.  I supposed I could spell out "in brackets" :)